API and Component Buildout for Homepage

sreya-kata opened the pull request:
Did the following:
- reorganized component file structure so every utilized component has its own folder with a js file and (future) css file
- modified homepage content model in CMS to coincide with new design
- data fetched all CMS components of the homepage into a graphql query
- made initial components for all distinct homepage components based on hi-fis (hero, mission, inclusivity w/carousel, stay in the know, and final carousel)

Note: from the code, "mission", "inclusivity", and "stay in the know" seem similar but separate react components are made as they are each paired with another feature (ie: inclusivity has a passive carousel, "stay in the know" has a featured image, the stylings slightly differ). See hi-fi representation below:
![hifi](https://user-images.githubusercontent.com/60420625/112706488-e4295380-8e7a-11eb-8dbe-db17e6707191.jpg)

sreya-kata committed:
added initial homepage features

sreya-kata committed:
added hero and section components

sreya-kata committed:
took out unused variable

sreya-kata committed:
finished data fetching for homepage

sreya-kata committed:
Merge branch 'main' of https://github.com/Scout-NU/vital-site into api-buildout/homepage

sreya-kata committed:
added almost all relevant components

sreya-kata committed:
added all basic homepage components and organized file structure

josephmannis commented on src/components/homepage/carousel/carousel.js:
```diff
@@ -0,0 +1,26 @@
+      <p>{title}</p>
+      {carouselItems.map(item => {
+        return (
+          <div>
```
🔑

josephmannis commented on src/components/homepage/carousel/carousel.js:
```diff
@@ -0,0 +1,26 @@
+          <div>
+            <h4>{item.title}</h4>
+            <p>{item.subTitle}</p>
+            <a href={item.link}>Learn More</a>
```
Can you please add target="_blank"

josephmannis replied:
Also, do these go to pages on the site or external links?

sreya-kata replied:
They're on the site. In the CMS, they're inputted as the name of page, but as the file name format (ie: "/get-involved")
Also I forgot to include the link in the query so adding that now^

josephmannis commented on src/components/homepage/inclusivity/inclusivity.js:
```diff
@@ -0,0 +1,22 @@
+    <>
+      <h2>{title}</h2>
+      <p>{blurbText.blurbText}</p>
+      <a href={ctaLink}>
```
🎯

josephmannis commented on src/components/homepage/inclusivity/inclusivity.js:
```diff
@@ -0,0 +1,22 @@
+      <h2>{title}</h2>
+      <p>{blurbText.blurbText}</p>
+      <a href={ctaLink}>
+        <button type="button">{ctaTitle}</button>
```
Can you please add an aria label here?

josephmannis commented on src/components/homepage/mission/mission.js:
```diff
@@ -0,0 +1,15 @@
+    <>
+      <h2>{title}</h2>
+      <p>{blurbText.blurbText}</p>
+      <a href={ctaLink}>
```
🎯 and aria label for the button, ty!

josephmannis commented on src/components/homepage/stay-in-the-know/stay-in-the-know.js:
```diff
@@ -0,0 +1,28 @@
+    <>
+      <h2>{title}</h2>
+      <p>{blurbText.blurbText}</p>
+      <a href={ctaLink}>
```
Same here

josephmannis commented on src/components/layout/layout.js:
```diff
@@ -29,7 +29,6 @@ const Layout = ({ children }) => {
       <div
         style={{
           margin: `0 auto`,
-          maxWidth: 960,
```
Can you pull out the inline styles please?

josephmannis commented on src/pages/index.js:
```diff
@@ -1,47 +1,103 @@
+        ctaLink={stayInTheKnow.ctaLink}
+        upcomingEvent={homePage.upcomingEvent}
+      />
+      <Carousel
```
This is very neatly organized!

josephmannis reviewed (changes requested):
Just a few things! Looks good otherwise

sreya-kata committed:
added href and aria-labels and added component for a single carousel item + added key for it

sreya-kata committed:
got rid of placeholder footer component in layout

sreya-kata commented:
- Added key to carousel item by also adding a separate react component for a single carousel item
- Added target and rel attributes to links
- Added aria-labels to buttons

josephmannis reviewed (approved):
There's a warning in the console for an unused import. Can you please clean that up, then everything should be good!

sreya-kata committed:
got rid of unused variable
